src/patient-registration/patient-registration.component.tsx

```diff
@@ -83,22 +90,99 @@ export const PatientRegistration: React.FC = () => {
     return () => abortController.abort();
   }, []);
 
+  useEffect(() => {
+    if (existingPatient) {
+      patientUuidMap['patientUuid'] = existingPatient.id;
+      // set names
+      if (existingPatient.name) {
+        existingPatient.name.forEach((name, index) => {
+          if (index === 0) {
+            patientUuidMap['preferredNameUuid'] = name.id;
+            initialFormValues.givenName = name.given[0];
+            initialFormValues.middleName = name.given[1];
+            initialFormValues.familyName = name.family;
+          }
+          if (index === 1) {
+            patientUuidMap['additionalNameUuid'] = name.id;
+            initialFormValues.addNameInLocalLanguage = true;
+            initialFormValues.additionalGivenName = name.given[0];
+            initialFormValues.additionalMiddleName = name.given[1];
+            initialFormValues.additionalFamilyName = name.family;
+          }
+        });
+      } else {
+        initialFormValues.unidentifiedPatient = true;
+      }
+      initialFormValues.gender = capitalize(existingPatient.gender);
+      initialFormValues.birthdate = existingPatient.birthDate;
+      initialFormValues.telephoneNumber = existingPatient.telecom ? existingPatient.telecom[0].value : '';
+
+      existingPatient.identifier.forEach(id => {
+        const key = id.system ? camelCase(id.system) : camelCase(id.type.text);
+        patientUuidMap[key] = {
+          uuid: id.id,
+          value: id.value,
+        };
+        initialFormValues[key] = id.value;
+      });
+
+      if (existingPatient.address && existingPatient.address[0]) {
+        const address = existingPatient.address[0];
+        Object.keys(address).forEach(prop => {
+          switch (prop) {
+            case 'id':
+              patientUuidMap['preferredAddressUuid'] = address[prop];
+              break;
+            case 'city':
```

Comment:
I think this should be more dynamic. This has the potential of missing address fields that were configured through the address template.

Reply:
Extra address fields that aren't supported by FHIR are handled from the `extension` property, here I handle them dynamically with same intensions. Checkout details from this slack [thread](https://openmrs.slack.com/archives/CKLPH66BB/p1599737968021100?thread_ts=1599571163.010700&cid=CKLPH66BB)